src/monitors/rocketMonitor.cpp

```diff
@@ -0,0 +1,175 @@
+        DM faceDM;
+        DM cellDM;
+
+        VecGetDM(faceGeomVec, &faceDM) >> checkError;
+        const PetscScalar* faceGeomArray;
+        VecGetArrayRead(faceGeomVec, &faceGeomArray) >> checkError;
+        VecGetDM(cellGeomVec, &cellDM) >> checkError;
+        const PetscScalar* cellGeomArray;
+        VecGetArrayRead(cellGeomVec, &cellGeomArray) >> checkError;
+
+        PetscReal tol = 1e-3;
+
+        // initialize vectors (send_buffer)
+        PetscReal mDotCell[3] = {0,0,0};
+        PetscReal mDotTotal[3] = {0,0,0};
+        PetscReal thrustCell[3] = {0,0,0};
+        PetscReal thrustTotal[3] = {0,0,0};
+
+        // initialize global vectors (receive_buffer)
+        PetscReal mDotTotalGlob[3] = {0,0,0};
+        PetscReal thrustTotalGlob[3] = {0,0,0};
+        PetscReal IspGlob[3] = {0,0,0};
+
+        // find all faces
+        PetscInt fStart, fEnd;
+        DMPlexGetHeightStratum(dm, 1, &fStart, &fEnd) >> checkError;
+
+        /** Get the current rank associated with this process */
+        PetscMPIInt rank;
+        MPI_Comm_rank(MPI_COMM_WORLD, &rank);  //!< Get the origin rank of the current process. The particle belongs to this rank. The rank only needs to be read once.
+
+
+        // need to look for faces at specified fieldBoundary then find cells bordering those faces which are in specified region
+        for (PetscInt face = fStart; face < fEnd; ++face) { // Iterate through all faces to check if in fieldBoundary
+            if (ablate::domain::Region::InRegion(monitor->fieldBoundary, dm, face)) { // Check if each face is in fieldBoundary
+                PetscFVFaceGeom* fg;
+                DMPlexPointLocalRead(faceDM, face, faceGeomArray, &fg) >> checkError; // read face geometry for face
+
+                PetscInt numberNeighborCells;
+                const PetscInt* neighborCells;
+                DMPlexGetSupportSize(dm, face, &numberNeighborCells) >> ablate::checkError;
+                DMPlexGetSupport(dm, face, &neighborCells) >> ablate::checkError;
+                for (PetscInt n = 0; n < numberNeighborCells; n++) {
+
+                    // Make sure that we are not working with a ghost cell
+                    PetscInt ghost = -1;
+                    if (ghostLabel) {
+                        DMLabelGetValue(ghostLabel, neighborCells[n], &ghost);
+                    }
+                    if (ghost >= 0) {
+                        continue;
+                    }
+
+                    if (ablate::domain::Region::InRegion(monitor->region, dm, neighborCells[n])) {  // check if cell is in region
+
+                        DMPlexPointLocalRead(solDM, neighborCells[n], solArray, &conservedValues);  // Retrieve conserved values from cell
+                        monitor->computePressure = monitor->eos->GetThermodynamicFunction(eos::ThermodynamicProperty::Pressure, monitor->GetSolver()->GetSubDomain().GetFields()); // get decode state function/context
```

Comment:
The get thermodynamic function can be expensive, would you mind moving this to the register with a class level variable?